### ISSUE TITLE
Searching for MobileNetV3

### ISSUE THREAD

**Issue by Arata2887** [Arata2887/issue_copy#5]
1.どんな論文? /What does this work do?

- https://arxiv.org/abs/1905.02244
- Present the MobileNetV3, which is tuned to mobile phone CPUs through a combination of hardware-aware network architecture search (NAS) complemented by the NetAdapt algorithm.
- Propose a new efficient segmentation decoder Lite Reduced Atrous Spatial Pyramid Pooling (LR-ASPP), achieve new state of the art results for mobile classification, detection and segmentation.

2.先行研究と比べてどこがすごい? /What makes this work greater than existing works?

![Figure 1](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/7a44a43e-b19e-4acb-84a3-681b1c06834d)

As can be seen on Figure 1, MobileNetV3-Large is 3.2% more accurate on ImageNet classification while reducing latency by 20% compared to MobileNetV2. MobileNetV3-Small is 6.6% more accurate compared to a MobileNetV2 model with comparable latency. Meanwhile, MobileNetV3 outperform the art at the time such as MnasNet, ProxylessNas.

![Table 6](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/97e86d7a-a3dc-4e37-9f07-ac9f64338cac)

In object detection, MobileNetV3-Large detection is over 25% faster at roughly the same accuracy as MobileNetV2 on COCO detection. MobileNetV3-Large LRASPP is 34% faster than MobileNetV2 R-ASPP at similar accuracy for Cityscapes segmentation.

3.技術や手法のキモはどこ? /What is the heart of this technology or method?

MobileNetV3 mainly improves MobileNetV2 from four aspects: **Squeeze-and-excite module** and **Update activation functions**, **Network Architecture Search**, and **Redesigning Expensive Layers**.

(1)	Squeeze-and-excite module

![Figure 4](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/6c5b15e2-d2b9-4255-aaae-0bf8d16ed19c)

MnasNet built upon the MobileNetV2 structure by introducing lightweight attention modules based on squeeze and excitation into the bottleneck structure. MobileNetV3 use a combination of similar layers as building blocks.

(2)	Update activation functions

MobileNetV2 uses ReLU6 as activation. However, In some mainstream models of the time, a nonlinearity called swish was introduced that when used as a drop-in replacement for ReLU, that significantly improves the accuracy of neural networks. The nonlinearity is defined as: 
![swish](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/7fdfaaeb-8ba2-44dd-ac18-3b033861041d)
But it comes with non-zero cost in embedded environments as the sigmoid function is much more expensive to compute on mobile devices.MobileNetV3 replace sigmoid function with its piece-wise linear hard analog: 
![h-sigmoid](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/ff78b824-4871-4b99-b1df-8cc1cb632ed4)
Similarly, the hard version of swish becomes:
![h-swish](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/0b76f423-f5da-4bc4-86d7-ebec00b81ed5)
The comparison of the soft and hard version of sigmoid and swish nonlinearities is shown in figure 6.

![Figure 6](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/4c7bcd12-0ffb-48bd-83ae-84482b36a92d)

> Figure 6: In practice, h-swish can be implemented as a piece-wise function to reduce the number of memory accesses driving the latency cost down substantially.
 
(3)	Network Architecture Search

For MobileNetV3, Google use platform-aware NAS to search for the global network structures by optimizing each network block, Then use the NetAdapt algorithm to search per layer for the number of filters.

(4)	Redesigning Expensive Layers

In contrast to MobileNetV2, in MobileNetV3, Google was able to reduce the number of filters to 16 while maintaining the same accuracy as 32 filters using either ReLU or swish. This saves an additional 2 milliseconds and 10 million MAdds.

![Figure 5](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/c9ed715a-908a-4785-86e3-d76f927d0c49)

The original and optimized last stages can be seen in figure 5. The efficient last stage reduces the latency by 7 milliseconds which is 11% of the running time and reduces the number of operations by 30 millions MAdds with almost no loss of accuracy.

4.どうやって有効だと検証した? /How is this work validated?

![Table 8](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/ecc73588-d8f0-4311-b5fa-3b89a960c33b)

Google reports results on classification(Figure 1), detection(Table 6) and segmentation(Table 8, the table above). They also report various ablation studies to shed light on the effects of various **design decisions**.

![Table 5](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/8b92d007-c94b-4dc5-b1a9-baeeaf00b028)

**Impact of non-linearities:** Table 5 show that using an optimized implementation of h-swish saves 6ms (more than 10% of the runtime). Optimized h-swish only adds an additional 1ms compared to traditional ReLU.

![Figure 9](https://github.com/jr-tagawalab-nttcom/paper-survey/assets/119403000/1603a67d-183e-492d-beda-71af02b7581d)

**Impact of other components:** Figure 9 show how introduction of different components moved along the latency/accuracy curve.

5.議論はある? /Any discussion for this work?

6.読んでいてわからなかったところは? /What don't you understand for this paper?

Network architecture searchについては、これまでNASに関する論文をほとんど読んだことがなかったので、あまり理解できませんでした。

7.公開コードやデータセットは? /Public codes or datasets?

https://github.com/pytorch/vision/blob/main/torchvision/models/mobilenetv3.py

8.次に読むべき論文は? /What should we read next?

https://arxiv.org/abs/2107.08430
YOLOX: Exceeding YOLO Series in 2021
